### PR TITLE
Add support for content URI trigger.

### PR DESCRIPTION
**Pull request by unEgor**

### CODE REVIEW

File: jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java [hunk 1]
```diff
@@ -153,6 +174,13 @@ private void encodeTrigger(JobTrigger trigger, Bundle data) {
                 t.getWindowStart());
             data.putInt(prefix + BundleProtocol.PACKED_PARAM_TRIGGER_WINDOW_END,
                 t.getWindowEnd());
+        } else if (trigger instanceof JobTrigger.ContentUriTrigger) {
+            data.putInt(
+                prefix + BundleProtocol.PACKED_PARAM_TRIGGER_TYPE,
+            BundleProtocol.TRIGGER_TYPE_CONTENT_URI);
```
Comment (ciarand) [inline jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java]: what happened to this formatting?

File: jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java [hunk 2]
```diff
@@ -185,4 +213,42 @@ private void encodeRetryStrategy(RetryStrategy retryStrategy, Bundle data) {
...
+            jsonFlags.put(uri.getFlags());
+            jsonUris.put(uri.getUri());
+        }
+        try {
+            contentUris.put(JSON_URI_FLAGS, jsonFlags);
+            contentUris.put(JSON_URIS, jsonUris);
+        } catch (JSONException e) {
+            throw new RuntimeException(e);
+        }
+      return contentUris.toString();
```
Comment (ciarand) [inline jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java]: nit: indentation is off

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayJobWriterTest.java [hunk 3]
```diff
@@ -235,6 +239,23 @@ public void testWriteToBundle_triggers() {
...
+        Bundle bundle = mWriter.writeToBundle(
+            initializeDefaultBuilder().setTrigger(contentUriTrigger).build(), new Bundle());
+        Uri[] uris =
+            (Uri[]) bundle.getParcelableArray(BundleProtocol.PACKED_PARAM_CONTENT_URI_ARRAY);
+        int[] flags = bundle.getIntArray(BundleProtocol.PACKED_PARAM_CONTENT_URI_FLAGS_ARRAY);
+        assertTrue("Array size", uris.length == flags.length && flags.length == 1);
+        assertEquals(BundleProtocol.PACKED_PARAM_CONTENT_URI_ARRAY,
+            uris[0], ContactsContract.AUTHORITY_URI);
+        assertEquals(BundleProtocol.PACKED_PARAM_CONTENT_URI_FLAGS_ARRAY,
+            Flags.FLAG_NOTIFY_FOR_DESCENDANTS, flags[0]);
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayJobWriterTest.java]: why are these reversed?

File: jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java (hunk 1 above)
Comment (unEgor) [inline jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java]: fixed

File: jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java (hunk 2 above)
Comment (unEgor) [inline jobdispatcher/src/main/java/com/firebase/jobdispatcher/JobCoder.java]: fixed

**Pull request merged**